sdk/tests/integration/ContractExecuteTransactionIntegrationTests.cc

```diff
@@ -0,0 +1,284 @@
+                 ContractExecuteTransaction()
+                   .setGas(100000ULL)
+                   .setFunction("setMessage", ContractFunctionParameters().addString("new message"))
+                   .execute(getTestClient())
+                   .getReceipt(getTestClient()),
+               ReceiptStatusException); // INVALID_CONTRACT_ID
+}
+
+//-----
+TEST_F(ContractExecuteTransactionIntegrationTest, CannotExecuteContractWithNoFunctionParameters)
+{
+  // Given
+  const std::unique_ptr<PrivateKey> operatorKey = ED25519PrivateKey::fromString(
+    "302e020100300506032b65700422042091132178e72057a1d7528025956fe39b0b847f200ab59b2fdd367017f3087137");
+  FileId fileId;
+  ASSERT_NO_THROW(fileId = FileCreateTransaction()
+                             .setKey(operatorKey->getPublicKey())
+                             .setContents(internal::HexConverter::hexToBytes(getTestSmartContractBytecode()))
+                             .execute(getTestClient())
+                             .getReceipt(getTestClient())
+                             .getFileId()
+                             .value());
+  ContractId contractId;
+  ASSERT_NO_THROW(contractId =
+                    ContractCreateTransaction()
+                      .setBytecodeFileId(fileId)
+                      .setAdminKey(operatorKey->getPublicKey())
+                      .setGas(100000ULL)
+                      .setConstructorParameters(ContractFunctionParameters().addString("Hello from Hedera.").toBytes())
+                      .execute(getTestClient())
+                      .getReceipt(getTestClient())
+                      .getContractId()
+                      .value());
+
+  // When / Then
+  EXPECT_THROW(const TransactionReceipt txReceipt = ContractExecuteTransaction()
+                                                      .setContractId(contractId)
+                                                      .setGas(100000ULL)
+                                                      .execute(getTestClient())
+                                                      .getReceipt(getTestClient()),
+               ReceiptStatusException); // CONTRACT_REVERT_EXECUTED
+
+  // Clean up
+  TransactionReceipt txReceipt;
+  ASSERT_NO_THROW(txReceipt = ContractDeleteTransaction()
+                                .setContractId(contractId)
+                                .setTransferAccountId(AccountId(2ULL))
+                                .execute(getTestClient())
+                                .getReceipt(getTestClient()));
+  ASSERT_NO_THROW(txReceipt =
+                    FileDeleteTransaction().setFileId(fileId).execute(getTestClient()).getReceipt(getTestClient()));
+}
+
+//-----
+TEST_F(ContractExecuteTransactionIntegrationTest, CannotExecuteContractWithNoGas)
+{
+  // Given
+  const std::unique_ptr<PrivateKey> operatorKey = ED25519PrivateKey::fromString(
+    "302e020100300506032b65700422042091132178e72057a1d7528025956fe39b0b847f200ab59b2fdd367017f3087137");
```

Comment:
Same here - can't we just use the private key from `local_node.json`?

Reply:
See previous comment.